src/libraries/System.Net.Http/src/System/Net/Http/ByteArrayContent.cs

```diff
@@ -58,6 +59,15 @@ public ByteArrayContent(byte[] content, int offset, int count)
         private protected Task SerializeToStreamAsyncCore(Stream stream, CancellationToken cancellationToken) =>
             stream.WriteAsync(_content, _offset, _count, cancellationToken);
 
+        protected override void SerializeToStream(Stream stream, TransportContext? context, CancellationToken cancellationToken)
+        {
+            Debug.Assert(stream != null);
+
+            // Last chance to check for timeout/cancellation, sync Stream API doesn't have any support for it.
```

Comment:
I don't know if we want to but we _could_ send in chunks here and check the token between chunks. This would not help exceptional cases, but would make large transfers somewhat cancellable, at the cost of smaller packets (we disable Nagle, so every write is a write...).

Of course if we update to dispose the socket, then we don't need any change here.

Reply:
Since this is `ByteArrayContent` and we don't need to worry about _reading_, and disposing the socket is handling the writing, I think this is a non-issue.